### PR TITLE
[PAYMENT] Add the ability to pay for car rental

### PR DESCRIPTION
**Pull request by HovorukhaBohdan**

### CODE REVIEW

File: src/main/java/mate/sep23/group3/car/sharing/repository/PaymentRepository.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import mate.sep23.group3.car.sharing.model.Payment;
+import mate.sep23.group3.car.sharing.model.Rental;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface PaymentRepository extends JpaRepository<Payment, Long> {
+    Page<Payment> findAllByRentalIn(List<Rental> rentals, Pageable pageable);
```
Comment (dima-semeniuk) [inline src/main/java/mate/sep23/group3/car/sharing/repository/PaymentRepository.java:15]: I think there should be findAllByRentalId

File: src/main/java/mate/sep23/group3/car/sharing/strategy/payment/handlers/roles/CustomerRoleHandler.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@RequiredArgsConstructor
+public class CustomerRoleHandler implements RoleHandler {
+    private final RentalRepository rentalRepository;
+    private final PaymentRepository paymentRepository;
+
+    @Override
+    public Page<Payment> getPayments(Long userId, Pageable pageable) {
+        List<Rental> rentals = rentalRepository.findAllByUserId(userId);
+
+        return paymentRepository.findAllByRentalIn(rentals, pageable);
```
Comment (dima-semeniuk) [inline src/main/java/mate/sep23/group3/car/sharing/strategy/payment/handlers/roles/CustomerRoleHandler.java:24]:
```suggestion
        return paymentRepository.findAllByRentalId(rentals, pageable);
```

**Review by dima-semeniuk — COMMENTED** [review]
Great job, but minor points to be considered

**Review by dima-semeniuk — APPROVED** [review]
Great job! Nothing to add

File: pom.xml [hunk 3]
```diff
@@ -110,11 +111,24 @@
             <version>2.3.1</version>
         </dependency>
 
+        <dependency>
+            <groupId>io.swagger.core.v3</groupId>
+            <artifactId>swagger-annotations-jakarta</artifactId>
+            <version>2.2.19</version>
+        </dependency>
```
Comment (ponomvrenko) [inline pom.xml]: Redundant. We have already swagger in our:
```
<dependency>
    <groupId>org.springdoc</groupId>
    <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
    <version>2.2.0</version>
</dependency>
```

**Review by ponomvrenko — CHANGES_REQUESTED** [review]
Good job! A small point needs to be made.

**Review by ponomvrenko — APPROVED** [review]
LGTM!

**Pull request merged**